vcd/resource_vcd_tm_org.go

```diff
@@ -0,0 +1,233 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/go-vcloud-director/v3/govcd"
+	"github.com/vmware/go-vcloud-director/v3/types/v56"
+)
+
+const labelTmOrg = "TM Organization"
+
+func resourceVcdTmOrg() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdTmOrgCreate,
+		ReadContext:   resourceVcdTmOrgRead,
+		UpdateContext: resourceVcdTmOrgUpdate,
+		DeleteContext: resourceVcdTmOrgDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdTmOrgImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
```

Comment:
Is it `ForceNew: true`? In UI is not allowed to update it. If I try with Terraform I get
```
│ Error: error updating TM Organization with ID: error updating entity of type 'Organization': error in HTTP PUT request: ACCESS_TO_RESOURCE_IS_FORBIDDEN - [ 4-2024-10-23-11-06-35-201--24120f6e-9687-4633-adf0-b146b7161152 ] Either you need some or all of the following rights [ORG_NAME_EDIT] to perform operations [ORGANIZATION_EDIT_NAME] for 06708dee-297e-435b-9675-bb373a1b8b85 or the target entity is invalid.
```

Reply:
So this is interesting. It does not allow naming when the Org is enabled, but API accepts rename in two cases:
* The Org was disabled and will be enabled with the same API call that the rename occurs in
* The Org was enabled, but will be disabled with the same API call that the rename occurs in

I have tuned manual validation to be more informative. The test that goes through both these cases `TestAccVcdTmOrg` passes.